services/core/DataMover/tests/test_datamover.py

```diff
@@ -0,0 +1,696 @@
+        order="LAST_TO_FIRST").get(timeout=10)
+    print('Query Result', result)
+    assert (len(result['values']) == 1)
+    assert (result['values'][0][1] == mixed_reading)
+
+
+@pytest.mark.historian
+@pytest.mark.forwarder
+def test_log_topic_no_header(publish_agent, query_agent):
+    """
+    Test if log topic message is getting forwarded to historian running on
+    another instance. Test if topic name substitutions happened.
+    Publish to topic
+    'datalogger/PNNL/BUILDING_1/Device' in volttron_instance1 and
+    query for topic
+    'datalogger/PNNL/BUILDING1_ANON/Device/MixedAirTemperature' in
+    volttron_instance2
+
+    @param publish_agent: Fake agent used to publish messages to bus in
+    volttron_instance1. Calling this fixture makes sure all the dependant
+    fixtures are called to setup and start volttron_instance1 and forwareder
+    agent and returns the  instance of fake agent to publish
+    @param query_agent: Fake agent used to query sqlhistorian in
+    volttron_instance2. Calling this fixture makes sure all the dependant
+    fixtures are called to setup and start volttron_instance2 and sqlhistorian
+    agent and returns the instance of a fake agent to query the historian
+    """
+
+    print("\n** test_log_topic **")
+    # Publish fake data. The format mimics the format used by VOLTTRON drivers.
+    # Make some random readings
+    oat_reading = random.uniform(30, 100)
+    mixed_reading = oat_reading + random.uniform(-5, 5)
+    current_time = datetime.utcnow().isoformat()
+    # Create a message for all points.
+    message = {'MixedAirTemperature': {'Readings': mixed_reading,
+                                       'Units': 'F',
+                                       'tz': 'UTC',
+                                       'type': 'float'}}
+    gevent.sleep(1)  # sleep so that there is no side effect from earlier test
+    # Publish messages
+    publish(publish_agent, "datalogger/PNNL/BUILDING_1/Device", None, message)
+    gevent.sleep(0.5)
+
+    # Query the historian
+    result = query_agent.vip.rpc.call(
+        'platform.historian',
+        'query',
+        topic="datalogger/PNNL/BUILDING1_ANON/Device/MixedAirTemperature",
+        start=current_time,
+        order="LAST_TO_FIRST").get(timeout=10)
+    print('Query Result', result)
+    assert (len(result['values']) == 1)
+    assert (result['values'][0][1] == mixed_reading)
+
+
+@pytest.mark.historian
+@pytest.mark.forwarder
+def test_topic_not_forwarded(publish_agent, query_agent):
+    """
+    Test if devices topic message is getting forwarded to historian running on
+    another instance. Test if topic name substitutions happened.
+    Publish to topic
+    'datalogger/PNNL/BUILDING_1/Device' in volttron_instance1 and
+    query for topic
+    'datalogger/PNNL/BUILDING1_ANON/Device/MixedAirTemperature' in
+    volttron_instance2
+
+    @param publish_agent: Fake agent used to publish messages to bus in
```

Comment:
I don't believe this is well formed sphinx code.  As it it is.  It's hard to read without the spaces below the wrapped txts of the params.